utils/load_path.mli

```diff
@@ -29,7 +29,10 @@ val remove_dir : string -> unit
 (** Remove a directory from the load path *)
 
 val reset : unit -> unit
-(** Remove all directories *)
+(** Remove all directories and caches. *)
+
+val reset_load_path : unit -> unit
```

Comment:
This name is confusing (what's the difference between `reset` and `reset_load_path`?).
A better name is needed. Maybe `clear`? (there is a slight analogy with the difference between `Hashtbl.reset` and `Hashtbl.clear`).

Reply:
Good idea, switched to `clear`, with a bit of documentation.